sormas-ui/src/main/java/de/symeda/sormas/ui/user/UserRoleNotificationCheckboxSet.java

```diff
@@ -0,0 +1,245 @@
+import com.vaadin.v7.ui.CustomField;
+
+import de.symeda.sormas.api.i18n.Captions;
+import de.symeda.sormas.api.i18n.I18nProperties;
+import de.symeda.sormas.api.i18n.Strings;
+import de.symeda.sormas.api.user.NotificationType;
+import de.symeda.sormas.api.user.NotificationTypeGroup;
+import de.symeda.sormas.api.user.UserRoleDto;
+import de.symeda.sormas.ui.utils.ButtonHelper;
+import de.symeda.sormas.ui.utils.CssStyles;
+import de.symeda.sormas.ui.utils.components.CheckboxSetItemDataSource;
+
+public class UserRoleNotificationCheckboxSet extends CustomField<UserRoleDto.NotificationTypes> {
+
+	private static final long serialVersionUID = 4149126643467960889L;
+
+	private VerticalLayout layout;
+
+	@Override
+	protected Component initContent() {
+		layout = new VerticalLayout();
+		layout.setWidthFull();
+		layout.setMargin(new MarginInfo(true, false));
+
+		buildCheckboxes();
+
+		return layout;
+	}
+
+	@Override
+	public Class<? extends UserRoleDto.NotificationTypes> getType() {
+		return UserRoleDto.NotificationTypes.class;
+	}
+
+	protected void setInternalValue(UserRoleDto.NotificationTypes newValue) {
+		super.setInternalValue(UserRoleDto.NotificationTypes.of(newValue.getSms(), newValue.getEmail()));
+	}
+
+	private void buildCheckboxes() {
+		for (NotificationTypeGroup group : NotificationTypeGroup.values()) {
+			List<CheckboxRow> rows = createGroupRows(group);
+
+			layout.addComponent(createGroupHeader(group, rows));
+			layout.addComponents(rows.toArray(new Component[] {}));
+		}
+	}
+
+	private HorizontalLayout createGroupHeader(NotificationTypeGroup group, List<CheckboxRow> rows) {
+		Label groupLabel = new Label(group.toString());
+		groupLabel.addStyleName(CssStyles.H3);
+
+		Label tickForAllLAbel = new Label(I18nProperties.getString(Strings.checkboxSetTickAnAnswerForAll));
+		tickForAllLAbel.addStyleName(CssStyles.LABEL_ITALIC);
+
+		Button tickAllButton = ButtonHelper.createButton(
+			I18nProperties.getCaption(Captions.all),
+			(e) -> rows.forEach(CheckboxRow::checkAll),
+			ValoTheme.BUTTON_LINK,
+			CssStyles.BUTTON_COMPACT);
+		Button tickSmsButton = ButtonHelper.createButton(
+			I18nProperties.getCaption(Captions.userRoleNotificationTypeSms),
+			(e) -> rows.forEach(CheckboxRow::checkAllSms),
+			ValoTheme.BUTTON_LINK,
+			CssStyles.BUTTON_COMPACT);
+		Button tickEmailButton = ButtonHelper.createButton(
+			I18nProperties.getCaption(Captions.userRoleNotificationTypeEmail),
+			(e) -> rows.forEach(CheckboxRow::checkAllEmail),
+			ValoTheme.BUTTON_LINK,
+			CssStyles.BUTTON_COMPACT);
+
+		HorizontalLayout buttonsLayout = new HorizontalLayout(tickForAllLAbel, tickAllButton, tickSmsButton, tickEmailButton);
+		buttonsLayout.setWidthFull();
+		buttonsLayout.setExpandRatio(tickForAllLAbel, 1);
+
+		HorizontalLayout headerLayout = new HorizontalLayout(groupLabel, buttonsLayout);
+		headerLayout.setWidthFull();
+
+		headerLayout.setExpandRatio(groupLabel, 0.5f);
+		headerLayout.setExpandRatio(buttonsLayout, 0.5f);
+
```

Comment:
Add `headerLayout.setComponentAlignment(buttonsLayout, Alignment.MIDDLE_RIGHT);` to have the "all" buttons aligned with the caption.